packages/eslint-plugin/lib/rules/prefer-icon-component.js

```diff
@@ -0,0 +1,70 @@
+const jsxAstUtils = require('jsx-ast-utils');
+
+const { elementType, getProp, getLiteralPropValue } = jsxAstUtils;
+
+const MESSAGE =
+  'The <va-icon> Web Component should be used instead of Font Awesome. See: https://design.va.gov/storybook/?path=/docs/uswds-va-icon--default';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    fixable: 'code',
+  },
+
+  create(context) {
+    return {
+      JSXElement(node) {
+        const iconNode = node.openingElement;
+        const classNameProp = getProp(iconNode.attributes, 'className');
+        const classValue = getLiteralPropValue(classNameProp);
+
+        // filter out font awesome classes
+        const classNameClasses = classValue
+          ?.split(' ')
+          ?.filter(c => !c.includes('fa'))
+          ?.join(' ');
+
+        const ariaLabelProp = getProp(iconNode.attributes, 'aria-label');
+        const ariaLabelText = getLiteralPropValue(ariaLabelProp);
+
+        const roleProp = getProp(iconNode.attributes, 'role');
+        const altProp = getProp(iconNode.attributes, 'alt');
+
+        if (classValue && classValue.includes('fa-')) {
+          context.report({
+            node,
+            message:  MESSAGE,
+            fix: fixer => {
+              return [
+                // replace the original element name with va-icon
+                fixer.replaceText(iconNode.name, 'va-icon'),
+
+                // add the size prop
+                fixer.insertTextAfter(iconNode.name, ` size={4}`),
+
+                //  add the icon prop
+                fixer.insertTextAfter(iconNode.name, ` icon="see Storybook for icon names: https://design.va.gov/storybook/?path=/docs/uswds-va-icon--default"`),
```

Comment:
Another place maybe we should link to the migration docs instead (to be written)?

Reply:
I'm going to keep the Storybook link for the `icon` prop here to make it easier to find the names. In the ESLint pop-up message, I added the other documentation link though.

Reply:
I like leaving the link for the `icon` prop so that it is easy to get the icon name

Reply:
I second what Ian said, with the extra link to clarify usage.